prov/efa/src/efa_device.c

```diff
@@ -97,6 +97,15 @@ int efa_device_construct(struct efa_device *efa_device,
 		goto err_close;
 	}
 
+	if (-FI_ENODATA == fi_param_get_int(&rxr_prov, "use_device_rdma", &tmp_value)) {
+		if (efa_device->ibv_attr.vendor_part_id < 0xefa2) {
+			rxr_env.use_device_rdma = 0;
+		} else {
+			assert(efa_device->efa_attr.device_caps & EFADV_DEVICE_ATTR_CAPS_RDMA_READ);
```

Comment:
Why is an assert right here?  Asserts aren't always compiled in and this seems less like an assumption enforced by code and more an assumption about the underlying system.  Seems better that to have always on error handling.  Also, asserts normally dump core, and this seems like a very wasteful error to dump core over.

Reply:
Thank you. I changed it into an error handling case in the latest revision.